### PR TITLE
ISSUE-1159: Add an API to delete toolbar metadata for a particular user

### PR DESCRIPTION
**Pull request by priyank5485**
Add an API to delete toolbar metadata for a particular user

### CODE REVIEW

**Comment by priyank5485** [comment]
@arunmahadevan I have created this PR but running in to issue while trying to delete. Debug logs as below. As discussed it could be a cache not being populated problem. But still putting this up for early review and unblock UI team so that they can make progress with UI related changes. Will keep looking in to fixing the cache related issue.

DEBUG  [2018-01-16 23:54:25.245] [dw-659 - DELETE /api/v1/catalog/system/topologyeditortoolbar/7] c.h.s.s.s.a.StreamlineKerberosRequestFilter -  Method: DELETE, AuthType: kerberos, RemoteUser: streamline_componentsndvFArfRg, UserPrincipal: u=streamline_componentsndvFArfRg&p=streamline_componentsndvFArfRg@HDF.COM&t=kerberos&e=1516182865181, Scheme: http
DEBUG  [2018-01-16 23:54:25.245] [dw-659 - DELETE /api/v1/catalog/system/topologyeditortoolbar/7] c.h.s.s.s.a.StreamlineKerberosRequestFilter -  SecurityContext StreamlineSecurityContext{principal=u=streamline_componentsndvFArfRg&p=streamline_componentsndvFArfRg@HDF.COM&t=kerberos&e=1516182865181, scheme='http', authenticationScheme='kerberos', isSecure=false}
DEBUG  [2018-01-16 23:54:25.251] [dw-659 - DELETE /api/v1/catalog/system/topologyeditortoolbar/7] c.h.s.s.s.i.DefaultStreamlineAuthorizer -  User: streamline_componentsndvFArfRg, Role: ROLE_ADMIN, Result: true
DEBUG  [2018-01-16 23:54:25.251] [dw-659 - DELETE /api/v1/catalog/system/topologyeditortoolbar/7] c.h.s.s.s.i.DefaultStreamlineAuthorizer -  DefaultStreamlineAuthorizer, AuthenticationContext: AuthenticationContext{, principal=u=streamline_componentsndvFArfRg&p=streamline_componentsndvFArfRg@HDF.COM&t=kerberos&e=1516182865181}, targetEntityNamespace: topology_editor_toolbar, targetEntityId: 7, permissions: [DELETE], result: true
WARN   [2018-01-16 23:54:25.254] [dw-659 - DELETE /api/v1/catalog/system/topologyeditortoolbar/7] c.h.r.s.CacheBackedStorageManager -  Possible cache inconsistency. Storable from DB 'com.hortonworks.streamline.streams.catalog.TopologyEditorToolbar@3018f277', Storable from cache 'null'

File: streams/service/src/main/java/com/hortonworks/streamline/streams/service/TopologyEditorToolbarResource.java [hunk 1]
```diff
@@ -116,6 +117,19 @@ public Response removeTopologyEditorToolbar(@Context SecurityContext securityCon
                 .orElseThrow(() -> EntityNotFoundException.byId(String.valueOf(userId)));
     }
 
+    @DELETE
+    @Path("/system/topologyeditortoolbar/{id}")
```
Comment (arunmahadevan) [inline streams/service/src/main/java/com/hortonworks/streamline/streams/service/TopologyEditorToolbarResource.java]: nit: id -> userId

**Comment by arunmahadevan** [comment]
@priyank5485 , not sure why was this PR raised only for HDF 3.1.0 branch. I have merged it to both master and HDF 3.1.0 and closing this.

**Comment by arunmahadevan** [comment]
@priyank5485 , can you make the necessary changes to `CacheBackedStorageManager` to avoid the NPE you observed while testing this patch ?

**Comment by priyank5485** [comment]
@arunmahadevan Tested that change and it worked. Raised a PR at https://github.com/hortonworks/registry/pull/390

**Pull request closed without merging**